### PR TITLE
Passing body data as JSON on POST for ServiceAccountTokenProvider

### PR DESCRIPTION
**Pull request by barrault01**
A small PR request to pass body data as a JSON and not URLComponents, it is not wrong to pass as URLComponents but I think it is much cleaner to pass as JSON.

### CODE REVIEW

**Comment by barrault01** [comment]
I signed it!

File: Sources/OAuth2/ServiceAccountTokenProvider.swift [hunk 1]
```diff
@@ -82,17 +82,13 @@ public class ServiceAccountTokenProvider : TokenProvider {
...
-    
+    let json: [String: Any] = ["grant_type": "urn:ietf:params:oauth:grant-type:jwt-bearer",
+                           "assertion": msg]
+    let data = try? JSONSerialization.data(withJSONObject: json)    
+  
     var urlRequest = URLRequest(url:URL(string:credentials.TokenURI)!)
     urlRequest.httpMethod = "POST"
-    urlRequest.httpBody = query.data(using:.utf8)
+    urlRequest.httpBody = data
     urlRequest.setValue("application/x-www-form-urlencoded", forHTTPHeaderField:"Content-Type")
```
[the remark this answers is not in the document]
Comment (barrault01) [reply Sources/OAuth2/ServiceAccountTokenProvider.swift]: I just update it!

**Comment by timburks** [comment]
Thank you!!

**Comment by barrault01** [comment]
@timburks are you planning to merge it?

**Comment by timburks** [comment]
@barrault01 yes, sorry to miss that

**Pull request merged**